src/databricks/labs/ucx/hive_metastore/migrate_tables.py

```diff
@@ -0,0 +1,61 @@
+import logging
+from functools import partial
+
+from databricks.sdk import WorkspaceClient
+
+from databricks.labs.ucx.framework.crawlers import SqlBackend
+from databricks.labs.ucx.framework.parallel import ThreadedExecution
+from databricks.labs.ucx.hive_metastore.tables import TablesCrawler
+
+logger = logging.getLogger(__name__)
+
+
+class TablesMigrate:
+    def __init__(
+        self,
+        tc: TablesCrawler,
+        ws: WorkspaceClient,
+        backend: SqlBackend,
+        default_catalog=None,
+        inventory_database="",
+        database_to_catalog_mapping: dict[str, str] | None = None,
+    ):
+        self._tc = tc
+        self._backend = backend
+        self._ws = ws
+        self._inventory_database = inventory_database
+        self._database_to_catalog_mapping = database_to_catalog_mapping
+        self._seen_tables = {}
+        self._default_catalog = self._init_default_catalog(default_catalog)
+
+    @staticmethod
+    def _init_default_catalog(default_catalog):
+        if default_catalog:
+            return default_catalog
+        else:
+            return "ucx_default"  # TODO : Fetch current workspace id
+
+    def migrate_tables(self):
+        tasks = []
+        if self._database_to_catalog_mapping:
```

Comment:
Embed this if into a loop, having it outside will cause people to upgrade both branches